plugin.yaml

```diff
@@ -10,58 +10,215 @@ plugins:
+          inputs:
+            gcp_config:
+              default: { get_property: [SELF, gcp_config] }
+
+  cloudify.gcp.nodes.KeyPair:
+    derived_from: cloudify.nodes.Root
+    properties:
+      use_external_resource:
+        description: >
+          Indicate whether the resource exists or if Cloudify should create the resource.
+        type: boolean
+        default: false
+      private_key_path:
+        description: >
+          The path where the key should be saved on the machine. If this is a bootstrap
+          process, this refers to the local computer. If this will run on the manager,
+          this will be saved on the manager.
+        type: string
+        default: ''
+      private_existing_key_path:
+        description: >
+          The path where the key should be saved on the machine. If this is a bootstrap
+          process, this refers to the local computer. If this will run on the manager,
+          this will be saved on the manager.
+        type: string
+        default: ''
+      public_existing_key_path:
+        description: >
+          The path to read from existing public key.
+        type: string
+        default: ''
+      gcp_config:
+        description: >
+          A dictionary of values to pass to authenticate with the Google Cloud Platform API.
+        default: {}
+      user:
+        type: string
+        default: ''
+    interfaces:
+      cloudify.interfaces.lifecycle:
+        create:
+          implementation: gcp_plugin.plugin.tasks.create_keypair
+          inputs:
+            gcp_config: 
+              default: { get_property: [SELF, gcp_config] }
+            user: 
+              default: { get_property: [SELF, user] }
+            private_key_path: 
+              default: { get_property: [SELF, private_key_path] }
+            private_existing_key_path: 
+              default: { get_property: [SELF, private_existing_key_path] }
+            public_existing_key_path: 
+              default: { get_property: [SELF, public_existing_key_path] }
+            external:
+              default: { get_property: [SELF, use_external_resource] }
+        delete:
+          implementation: gcp_plugin.plugin.tasks.delete_keypair
+          inputs:
+            gcp_config: 
+              default: { get_property: [SELF, gcp_config] }
+            user: 
+              default: { get_property: [SELF, user] }
+            private_key_path: 
+              default: { get_property: [SELF, private_key_path] }
+
+  cloudify.gcp.nodes.ExternalIP:
+    derived_from: cloudify.nodes.VirtualIP
+    properties:
+      use_external_resource:
+        description: >
+          Indicate whether the resource exists or if Cloudify should create the resource.
+        type: boolean
+        default: false
+      gcp_config:
+        description: >
+          A dictionary of values to pass to authenticate with the GCP API.
+        default: {}
+
 relationships:
   cloudify.gcp.relationships.connected_to:
```

Comment:
The relationships types "cloudify.gcp.relationships.connected_to" and "cloudify.gcp.relationships.contained_in" are not used, so you can remove them